src/daemon/distribution.c

```diff
@@ -0,0 +1,148 @@
+#include <math.h>
+#include "collectd.h"
+#include "distribution.h"
+
+struct bucket_s{
+	size_t bucket_counter;
```

Comment:
This counter here should be an `uint64_t`.

Rule of thumb: if it's used as an array index or as an argument to `calloc` / `malloc`, use `size_t`. If it's used as a counter metric, use `uint64_t`.

Reply:
Thank you for explaining.